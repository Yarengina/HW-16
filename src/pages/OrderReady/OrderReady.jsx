// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import { useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import UserInfoContext from '../../contexts/UserInfoContext'
import OrderInfoContext from '../../contexts/OrderInfoContext'

function OrderReady() {
    const { user } = useContext(UserInfoContext)
    const { order } = useContext(OrderInfoContext)

    const navigate = useNavigate()

    const goBack = () => {
        navigate('/order')
    }

    return (
        <div className="BookDesc">
            <h1>Receipt</h1>
            <p>
                Name: <strong>{user.name}</strong>
            </p>
            <p>
                Email: <strong>{user.email}</strong>
            </p>
            <p>
                Phone: <strong>{user.phone}</strong>
            </p>
            <p>
                Total Quantity: <strong>{order.quantity}</strong>
            </p>
            <p>
                Total Price: <strong>{order.price} руб.</strong>
            </p>
            <button className="ButtonBack" type="button" onClick={goBack}>
                Back to cart
            </button>
        </div>
    )
}

export default OrderReady
