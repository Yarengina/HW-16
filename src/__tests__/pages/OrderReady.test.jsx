// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import { render } from '../utils/renderWithRouter'
import OrderReady from '../../pages/OrderReady/OrderReady'

describe('<OrderReady />', () => {
    it('should render successfully and match snapshot', () => {
        const { container } = render(<OrderReady />)
        expect(container).toMatchSnapshot()
    })
})
