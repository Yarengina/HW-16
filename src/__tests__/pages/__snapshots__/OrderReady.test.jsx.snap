// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderReady /> should render successfully and match snapshot 1`] = `
<div>
  <div
    class="BookDesc"
  >
    <h1>
      Receipt
    </h1>
    <p>
      Name: 
      <strong />
    </p>
    <p>
      Email: 
      <strong />
    </p>
    <p>
      Phone: 
      <strong />
    </p>
    <p>
      Total Quantity: 
      <strong />
    </p>
    <p>
      Total Price: 
      <strong>
         руб.
      </strong>
    </p>
    <button
      class="ButtonBack"
      type="button"
    >
      Back to cart
    </button>
  </div>
</div>
`;
